### PR TITLE
Report qualname of exception type rather than <class 'RuntimeError'>

### PR DESCRIPTION
**Pull request by ezyang**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #135146
* #135145
* #135148

Signed-off-by: Edward Z. Yang <ezyang@meta.com>

cc @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @chenyang78 @kadeng @chauhang @amjames @rec

### CODE REVIEW

**Review by Skylion007 — APPROVED** [review]

File: torch/_dynamo/convert_frame.py [hunk 1]
```diff
@@ -915,7 +915,7 @@ def format_guard_failures() -> str:
             guarded_code = compile_inner(code, one_graph, hooks, transform)
             return guarded_code
         except Exception as e:
-            fail_type = str(type(e))
+            fail_type = type(e).__qualname__
             fail_reason = str(e)
```
Comment (albanD) [inline torch/_dynamo/convert_frame.py:919]: Aren't both of these lines dead code though? :thinking:

**Review by albanD — APPROVED** [review]
.

**Review by yanboliang — APPROVED** [review]

File: torch/_dynamo/convert_frame.py (hunk 1 above)
Comment (ezyang) [reply torch/_dynamo/convert_frame.py:919]: They're not, because they're used in the `finally` block

**Pull request closed without merging**